packages/framework/src/Model/Order/PlaceOrderFacade.php

```diff
@@ -117,15 +118,53 @@ public function createOrderOnly(OrderData $orderData): Order
      */
     protected function fillOrderItems(Order $order, OrderData $orderData): void
     {
+        // the collection is used to keep track of which items have already been created
+        $allOrderItems = new ArrayCollection($orderData->items);
```

Comment:
maybe a better name would be `$(all)OrderItemsToBeCreated`? or something like that? I got that idea after reading the line bellow where the newly created element is removed from the collection.

Reply:
another idea, is not it possible to keep track just by using the cache (`$alreadyCreatedOrderItems`)? I was having quite a hard time understanding the following code, especially this part can be confusing:
```
if (!$allOrderItems->contains($relatedOrderItemData)) {
    $relatedOrderItem = $alreadyCreatedOrderItems[$this->generateCacheKey($relatedOrderItemData)];
    $orderItem->addRelatedItem($relatedOrderItem);

    continue;
}
```

If we had just that one array, it could look like this:
```
$key = $this->generateCacheKey($relatedOrderItemData);
if (array_key_exists($key, $alreadyCreatedOrderItems, true)) {
    $orderItem->addRelatedItem($alreadyCreatedOrderItems[$key)]);

    continue;
}
```

I'll be glad to discuss it

Reply:
good idea – changed, thank 👍